src/modules/ce/components/ReferralPage.tsx

```diff
@@ -80,7 +81,17 @@ const ReferralPage: React.FC<Props> = ({}) => {
             <Stack direction='row' alignItems='center' gap={0.5}>
               <Person sx={{ color: 'grayscale.main' }} />
               <Typography variant='body1'>
-                {clientNameFromRecordWithOptionalClient(referral)}
+                {referral.client ? (
+                  <RouterLink
+                    to={generateSafePath(ClientDashboardRoutes.REFERRALS, {
+                      clientId: referral.client.id,
```

Comment:
This feels kind of unnecessarily tricky because we will need to check whether the user has permission to view this page in particular. Lets just link to the client dashboard overview page, which we know the user can see if the `referral.client` is present.